src/snowflake/cli/_app/telemetry.py

```diff
@@ -81,6 +86,33 @@ class TelemetryEvent(Enum):
 TelemetryDict = Dict[Union[CLITelemetryField, TelemetryField], Any]
 
 
+def _is_cli_exception(exception: Exception) -> bool:
+    return issubclass(
+        exception.__class__,
+        (click.ClickException, typer.Exit, typer.Abort, BrokenPipeError),
+    )
+
+
+def _get_additional_exception_information(exception: Exception) -> TelemetryDict:
+    """
+    Attach the errno and sqlstate if the exception or the
+    cause of the exception is a ProgrammingError
+    """
+    additional_info = {}
+
+    if isinstance(exception, ProgrammingError):
+        additional_info[CLITelemetryField.ERROR_CODE] = exception.errno
+        additional_info[CLITelemetryField.SQL_STATE] = exception.sqlstate
+
+    if isinstance(exception.__cause__, ProgrammingError):
```

Comment:
this will override the fields if the exception is a `ProgrammingError` and so is the cause, is that what we want?

Reply:
Yeah, good catch. IMHO the logic should be:

```
if programming error:
    log errno/sqlstate
if cause:
    log cause
    if cause is programming error:
       log cause's errno/sqlstate if not already present
```